### ISSUE TITLE
Obfuscated model from save total can't be opened

### ISSUE THREAD

**Issue by matteodepascali** [OpenModelica/OpenModelica#10513]
### Description
After exporting a model through the save total feature selecting the "obfuscate output" option, the resulting file can't be loaded. The following error message appears:

"[~/Desktop/testObfuscated/TestObfuscatedTotal.mo:1:6-1:6:writable] Error: Lexer failed to recognize '$n0
  Real'
" 

### Steps to reproduce
In the attached folder you may find the original model, the total obfuscated model and a launch script "test.mos". Running test.mos produces the reported behavior. 

### Expected behavior
omc should be able to load the model

### Additional files
Providing a zip folder with the files. 
[testObfuscated.zip](https://github.com/OpenModelica/OpenModelica/files/11177465/testObfuscated.zip)

### Version and OS
 - OpenModelica Version v1.22.0-dev-17-g88d7b964fd (64-bit)
 - Modelica 4.0.0
 - OS: Windows 11 64bit

**Comment by casella** [comment]
@perost this looks like a regression. We should check if it also happens in 1.21.0 and fix it there as well.

**Comment by perost** [comment]
Fixed in #10522, and I will port it to the 1.21.0 maintenance branch as well.

The issue was that we added some features to the obfuscation a while ago so that some parts of the model can be left as is, which means that there's a possibility of getting conflicting names if the model already contains names with the pattern n123.

So to avoid that I changed it to prefix the names with $, which is how we usually deal with compiler-generated names that don't need to be valid Modelica identifiers. Except that in this case they obviously need to be valid if you want to save the model and not just simulate it immediately (in order to obfuscate the generated C-code).

For now I just changed it back to not prefix the names with $, since it's unlikely to actually cause issues with normal models. To solve the issue properly we'd have to go through the whole model and make sure we don't generate identifiers that already exist in the model, but that will require a bit of work.
